### ISSUE TITLE
Deleting deployment won't delete replicaset and associated pod on AKS 1.19.3

### ISSUE THREAD

**Issue by cccs-eric** [Azure/AKS#1991]
**What happened**:
I couldn't confirm that this started to happen since the migration to 1.19.3, but I will try.  Meanwhile, here is the issue.  When I apply a simple deployment and then delete it, the replicaset and their associated pods are not being removed (while deployment, services, etc.. are).

**What you expected to happen**:
When removing a deployment, the associated replicaset and pods should also be removed.

**How to reproduce it (as minimally and precisely as possible)**:
I created a [script](https://gist.github.com/cccs-eric/38c43465778f294176ce33deb484f7ca) that exposes the problem.  Here is the output of that script when I run it:
```
Client Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.3", GitCommit:"1e11e4a2108024935ecfcb2912226cedeafd99df", GitTreeState:"clean", BuildDate:"2020-10-14T12:50:19Z", GoVersion:"go1.15.2", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.3", GitCommit:"1e11e4a2108024935ecfcb2912226cedeafd99df", GitTreeState:"clean", BuildDate:"2020-10-16T20:43:08Z", GoVersion:"go1.15.2", Compiler:"gc", Platform:"linux/amd64"}
```
```
+ kubectl create namespace delete-test
namespace/delete-test created
```
```
+ kubectl -n delete-test create deploy nginx --image nginx
deployment.apps/nginx created
```
```
+ kubectl -n delete-test get all
NAME                         READY   STATUS    RESTARTS   AGE
pod/nginx-6799fc88d8-29zmh   1/1     Running   0          5s

NAME                    READY   UP-TO-DATE   AVAILABLE   AGE
deployment.apps/nginx   1/1     1            1           5s

NAME                               DESIRED   CURRENT   READY   AGE
replicaset.apps/nginx-6799fc88d8   1         1         1       5s
```
```
+ kubectl -n delete-test delete deploy nginx --cascade=true
deployment.apps "nginx" deleted
```
```
+ kubectl -n delete-test get all
NAME                         READY   STATUS    RESTARTS   AGE
pod/nginx-6799fc88d8-29zmh   1/1     Running   0          11s

NAME                               DESIRED   CURRENT   READY   AGE
replicaset.apps/nginx-6799fc88d8   1         1         1       11s
```
```
+ kubectl delete namespace delete-test
namespace "delete-test" deleted
```

**Anything else we need to know?**:
I have 3 other AKS clusters running 1.18.8 and they do not have that issue.  I can run the same script on them and everything works as expected.

**Environment**:
- Kubernetes version (use `kubectl version`): 
```
Client Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.3", GitCommit:"1e11e4a2108024935ecfcb2912226cedeafd99df", GitTreeState:"clean", BuildDate:"2020-10-14T12:50:19Z", GoVersion:"go1.15.2", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.3", GitCommit:"1e11e4a2108024935ecfcb2912226cedeafd99df", GitTreeState:"clean", BuildDate:"2020-10-16T20:43:08Z", GoVersion:"go1.15.2", Compiler:"gc", Platform:"linux/amd64"}
```
- Size of cluster (how many worker nodes are in the cluster?)
- General description of workloads in the cluster (e.g. HTTP microservices, Java app, Ruby on Rails, machine learning, etc.)
- Others:

**Comment by cccs-eric** [comment]
I have confirmed that the issue is not related to k8s 1.19.3, but rather localized to my AKS.  I used a cluster at version 1.18.8, tested that all was fine, then upgraded it to 1.19.3 and it was still fine after.  So there must be some setting on my cluster that prevents the proper deletion.  

I understand that it is not a problem related to AKS in general, but can someone provide any insights on where to look for?
